Mathlib/LinearAlgebra/BilinearForm/Basic.lean

```diff
@@ -164,62 +176,76 @@ theorem add_apply (x y : M) : (B + D) x y = B x y + D x y :=
 multiplication.
 
 When `R` itself is commutative, this provides an `R`-action via `Algebra.id`. -/
+@[deprecated]
 instance {α} [Monoid α] [DistribMulAction α R] [SMulCommClass R α R] : SMul α (BilinForm R M) :=
   inferInstance
 
 #noalign bilin_form.coe_smul
 #noalign bilin_form.smul_apply
 
+@[deprecated]
 instance {α β} [Monoid α] [Monoid β] [DistribMulAction α R] [DistribMulAction β R]
     [SMulCommClass R α R] [SMulCommClass R β R] [SMulCommClass α β R] :
     SMulCommClass α β (BilinForm R M) := inferInstance
 
+@[deprecated]
 instance {α β} [Monoid α] [Monoid β] [SMul α β] [DistribMulAction α R] [DistribMulAction β R]
     [SMulCommClass R α R] [SMulCommClass R β R] [IsScalarTower α β R] :
     IsScalarTower α β (BilinForm R M) := inferInstance
 
+@[deprecated]
 instance {α} [Monoid α] [DistribMulAction α R] [DistribMulAction αᵐᵒᵖ R]
     [SMulCommClass R α R] [IsCentralScalar α R] :
     IsCentralScalar α (BilinForm R M) := inferInstance
 
+@[deprecated]
 instance : AddCommMonoid (BilinForm R M) := inferInstance
 
+@[deprecated]
 instance : Neg (BilinForm R₁ M₁) := inferInstance
 
```

Comment:
Deprecating instances is probably not a good idea; I think these should just be deleted.

Reply:
#12122